File: day2/solution1.py
dic = {"X": 1, "Y": 2, "Z": 3, "A": 1, "B": 2, "C": 3}


def score(a, b):
    if (dic[a] == dic[b]):
        return 3 + dic[b]
    elif (abs(dic[a]-dic[b]) < 2):
        return dic[b] if dic[a] > dic[b] else 6 + dic[b]
    return 6+dic[b] if dic[a] > dic[b] else dic[b]


with open('input.txt') as f:
    lines = [x.strip().split(" ") for x in f.readlines()]
    print(sum(list(map(lambda x: score(x[0], x[1]), lines))))

File: 2024/day1/solution1.rb
input_list = File.read("input.txt").split("\n").map(&:split)

left_numbers = input_list.map(&:first).map(&:to_i).sort
right_numbers = input_list.map(&:last).map(&:to_i).sort

puts left_numbers.zip(right_numbers).map { (_1 - _2).abs }.sum
